apps/gallery/test/marionette/activity_test.js

```diff
@@ -0,0 +1,130 @@
+/* global require, marionette, setup, __dirname */
+'use strict';
+
+var Gallery = require('./lib/gallery.js'),
+    GalleryActivityCaller = require('./lib/fakeactivitycaller.js'),
+    assert = require('assert');
+
+marionette('Gallery Activity Tests', function() {
+
+  var client, apps = {};
+
+  apps[GalleryActivityCaller.ORIGIN] = __dirname + '/apps/fakeactivitycaller';
+
+  client = marionette.client({
+    prefs: {
+      'device.storage.enabled': true,
+      'device.storage.testing': true,
+      'device.storage.prompt.testing': true
+    },
+
+    apps: apps
+  });
+
+  var galleryApp, activityCallerApp;
+  var imageInfo = {
+      name: 'pictures/firefoxOS.png',
+      type: 'image/png'
+    };
+
+  setup(function() {
+    // Remove all files in temp device storage.
+    client.fileManager.removeAllFiles();
+    // Add file into the pictures directory
+    client.fileManager.add({
+      type: 'pictures',
+      filePath: 'test_media/Pictures/firefoxOS.png'
+    });
+    galleryApp = new Gallery(client);
+    activityCallerApp = GalleryActivityCaller.create(client);
+  });
+
+  test('pick image using gallery pick activity', function() {
+    activityCallerApp.launch();
+    // Initiate pick activity by clicking pick image button
+    activityCallerApp.pickImage();
+
+    // Select image from thumbnail list in gallery app
+    galleryApp.selectImage();
```

Comment:
Do you need to do any kind of wait here? Is there a race condition or do we know that the gallery app has been launched?

If something goes wrong with launching the activity (e.g., if activityCallerApp initiates an "ick" activity instead of a "pick" activitiy) will we fail usefully here?  That is, do you want to have some sort of assertion here that the gallery app has actually opened up?